<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
<html xmlns="http://www.w3.org/TR/REC-html40" xmlns:v = 
"urn:schemas-microsoft-com:vml" xmlns:o = 
"urn:schemas-microsoft-com:office:office" xmlns:w = 
"urn:schemas-microsoft-com:office:word"><head><title>Software list</title>
<meta http-equiv="Content-Type" content="text/html; charset=unicode">
<meta content="Word.Document" name="ProgId">
<meta content="MSHTML 6.00.6000.17114" name="GENERATOR">
<meta content="Microsoft Word 10" name="Originator"><link 
href="software_files/filelist.xml" rel=File-List><link 
href="software_files/editdata.mso" rel=Edit-Time-Data><!--[if !mso]>
<style>
v\:* {behavior:url(#default#VML);}
o\:* {behavior:url(#default#VML);}
w\:* {behavior:url(#default#VML);}
.shape {behavior:url(#default#VML);}
</style>
<![endif]--><!--[if gte mso 9]><xml>
 <o:documentproperties>
  <o:author>rmjdjhz</o:author>
  <o:lastauthor>rmjdjhz</o:lastauthor>
  <o:revision>6</o:revision>
  <o:totaltime>10</o:totaltime>
  <o:created>2005-03-15T10:58:00Z</o:created>
  <o:lastsaved>2005-04-05T09:38:00Z</o:lastsaved>
  <o:pages>1</o:pages>
  <o:words>1301</o:words>
  <o:characters>7417</o:characters>
  <o:company>University College London</o:company>
  <o:lines>61</o:lines>
  <o:paragraphs>17</o:paragraphs>
  <o:characterswithspaces>8701</o:characterswithspaces>
  <o:version>10.6714</o:version>
 </o:documentproperties>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:worddocument>
  <w:zoom>108</w:zoom>
  <w:compatibility>
   <w:applybreakingrules/>
   <w:usefelayout/>
  </w:compatibility>
  <w:browserlevel>MicrosoftInternetExplorer4</w:browserlevel>
 </w:worddocument>
</xml><![endif]-->
<style>
<!--
 /* Font Definitions */
 @font-face
	{font-family:Wingdings;
	panose-1:5 0 0 0 0 0 0 0 0 0;
	mso-font-charset:2;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:0 268435456 0 0 -2147483648 0;}
@font-face
	{font-family:SimSun;
	panose-1:2 1 6 0 3 1 1 1 1 1;
	mso-font-alt:\5B8B\4F53;
	mso-font-charset:134;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 135135232 16 0 262145 0;}
@font-face
	{font-family:"\@SimSun";
	panose-1:2 1 6 0 3 1 1 1 1 1;
	mso-font-charset:134;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 135135232 16 0 262145 0;}
 /* Style Definitions */
 p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-parent:"";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman";
	mso-fareast-font-family:SimSun;}
h2
	{mso-margin-top-alt:auto;
	margin-right:0cm;
	mso-margin-bottom-alt:auto;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	mso-outline-level:2;
	font-size:18.0pt;
	font-family:"Times New Roman";
	font-weight:bold;}
a:link, span.MsoHyperlink
	{color:blue;
	text-decoration:underline;
	text-underline:single;}
a:visited, span.MsoHyperlinkFollowed
	{color:blue;
	text-decoration:underline;
	text-underline:single;}
p
	{mso-margin-top-alt:auto;
	margin-right:0cm;
	mso-margin-bottom-alt:auto;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman";
	mso-fareast-font-family:SimSun;}
@page Section1
	{size:595.3pt 841.9pt;
	margin:72.0pt 90.0pt 72.0pt 90.0pt;
	mso-header-margin:35.4pt;
	mso-footer-margin:35.4pt;
	mso-paper-source:0;}
div.Section1
	{page:Section1;}
 /* List Definitions */
 @list l0
	{mso-list-id:448355496;
	mso-list-template-ids:-655738518;}
@list l0:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Symbol;}
@list l0:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:"Courier New";
	mso-bidi-font-family:"Times New Roman";}
@list l0:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Wingdings;}
@list l1
	{mso-list-id:1299258185;
	mso-list-template-ids:-925706858;}
@list l1:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Symbol;}
@list l1:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:"Courier New";
	mso-bidi-font-family:"Times New Roman";}
@list l2
	{mso-list-id:1562406602;
	mso-list-template-ids:-209702584;}
@list l2:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Symbol;}
@list l2:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:"Courier New";
	mso-bidi-font-family:"Times New Roman";}
@list l2:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Wingdings;}
@list l2:level4
	{mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l2:level5
	{mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l2:level6
	{mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l2:level7
	{mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l2:level8
	{mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l2:level9
	{mso-level-tab-stop:324.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l3
	{mso-list-id:1897355396;
	mso-list-template-ids:853463830;}
@list l3:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:Symbol;}
@list l3:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	mso-ansi-font-size:10.0pt;
	font-family:"Courier New";
	mso-bidi-font-family:"Times New Roman";}
@list l3:level3
	{mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l3:level4
	{mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l3:level5
	{mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l3:level6
	{mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l3:level7
	{mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l3:level8
	{mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l3:level9
	{mso-level-tab-stop:324.0pt;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
ol
	{margin-bottom:0cm;}
ul
	{margin-bottom:0cm;}
-->
  </style>
<!--[if gte mso 10]>
<style>
 /* Style Definitions */
 table.MsoNormalTable
	{mso-style-name:"Table Normal";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-parent:"";
	mso-padding-alt:0cm 5.4pt 0cm 5.4pt;
	mso-para-margin:0cm;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";}
</style>
<![endif]-->
<meta content="Jing" name="Author"><!--[if gte mso 9]><xml>
 <o:shapedefaults v:ext="edit" spidmax="5122"/>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <o:shapelayout v:ext="edit">
  <o:idmap v:ext="edit" data="1"/>
 </o:shapelayout></xml><![endif]--></head>
<body lang="EN-US" vlink="blue" link="blue" bgcolor="white">
<div class="Section1">
<p class="Section1"><b><span lang="EN-US" 
style="FONT-SIZE: 18pt; FONT-FAMILY: Arial">C programs</span></b></p>
<p>2LD, EH+, FASTEH+, GENECOUNTING, HAP can be put in the same directory with 
call to the appropriate executables. EH+, FASTEH+ and GENECOUNTING can also 
share the same parameter and data file. I have tested them extensively under 
Windows, Sun, Dec Alpha and Linux systems. A brief comparison is shown in the 
following table. Program MCETDT (Monte Carlo module for the Extended 
Transmission Disequilibrium Test) is distributed with ETDT and unavailable from 
this site. <font color="#ff0000">Note to enter MS-DOS under the default setup for 
Windows Visa, you need to run c:\windows\system32\cmd.exe first</font>. </p>
<table class="MsoNormalTable" style="WIDTH: 638.5pt" height="289" cellpadding="0" 
width=851 border=1>
  <tbody>
  <tr style="HEIGHT: 17.25pt">
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 17.36%; PADDING-TOP: 0cm; HEIGHT: 17.25pt" 
    width="17%">
      <p class="MsoNormal"><b><span style="FONT-FAMILY: Arial">Package 
      name</span></b></p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 17.22%; PADDING-TOP: 0cm; HEIGHT: 17.25pt" 
    width="17%">
      <p class="MsoNormal"><b><span style="FONT-FAMILY: Arial">Executable 
      names</span></b></p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 15.24%; PADDING-TOP: 0cm; HEIGHT: 17.25pt" 
    width="15%">
      <p class="MsoNormal"><b><span 
      style="FONT-FAMILY: Arial">Documentation<o:p></o:p></span></b></p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 25.14%; PADDING-TOP: 0cm; HEIGHT: 17.25pt" 
    width="25%">
      <p class="MsoNormal"><b><span 
      style="FONT-FAMILY: Arial">Features</span></b></p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 23.64%; PADDING-TOP: 0cm; HEIGHT: 17.25pt" 
    width="23%">
      <p class="MsoNormal"><b><span 
      style="FONT-FAMILY: Arial">Limitations</span></b></p></td></tr>
  <tr style="HEIGHT: 18pt">
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 17.36%; PADDING-TOP: 0cm; HEIGHT: 18pt" 
    width="17%">
      <p class="MsoNormal">2LD</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 17.22%; PADDING-TOP: 0cm; HEIGHT: 18pt" 
    width="17%">
      <p class="MsoNormal">2ld</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 15.24%; PADDING-TOP: 0cm; HEIGHT: 18pt" 
    vAlign=top width="15%">
      <p class="MsoNormal"><o:p>&nbsp;</o:p></p>
      <p class="MsoNormal">2ld.doc, 2ld.htm</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 25.14%; PADDING-TOP: 0cm; HEIGHT: 18pt" 
    width="25%">
      <p class="MsoNormal">D', SE(D'), Cramer's V for two multiallelic loci, 
    r</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 23.64%; PADDING-TOP: 0cm; HEIGHT: 18pt" 
    width="23%">
      <p class="MsoNormal">Requires LDSHELL for many markers, does not show D' in 
      graphics</p></td></tr>
  <tr style="HEIGHT: 45.75pt">
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 17.36%; PADDING-TOP: 0cm; HEIGHT: 45.75pt" 
    width="17%">
      <p class="MsoNormal">&nbsp;</p>
      <p class="MsoNormal">EH+ </p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 17.22%; PADDING-TOP: 0cm; HEIGHT: 45.75pt" 
    width="17%">
      <p class="MsoNormal">&nbsp;</p>
      <p class="MsoNormal">eh, ehplus, fehp, pm, pmplus</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 15.24%; PADDING-TOP: 0cm; HEIGHT: 45.75pt" 
    vAlign=top width="15%">
      <p class="MsoNormal"><o:p>&nbsp;</o:p></p>
      <p class="MsoNormal">&nbsp;</p>
      <p class="MsoNormal"><o:p>&nbsp;</o:p></p>
      <p class="MsoNormal">pm.doc, pm.stm</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 25.14%; PADDING-TOP: 0cm; HEIGHT: 45.75pt" 
    width="25%">
      <p class="MsoNormal">Model-free and permutation tests of allelic 
      association. EH compatible and very easy to use. (having been<a 
      href="http://content.karger.com/ProdukteDB/produkte.asp?Aktion=hitparade&amp;ProduktNr=224250"> 
      the top paper in Hum Hered in 2005/6/7</a>). Refman library as of Feb 2005 
      <a href="software/ehplus.rmd">rmd</a> and <a 
      href="software/ehplus.rmx">rmx</a>)</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 23.64%; PADDING-TOP: 0cm; HEIGHT: 45.75pt" 
    width="23%">
      <p class="MsoNormal">Slower than FASTEH+ and GENECOUNTING but call FASTEH+ 
      for permutation tests from version 1.2</p></td></tr>
  <tr style="HEIGHT: 32.25pt">
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 17.36%; PADDING-TOP: 0cm; HEIGHT: 32.25pt" 
    width="17%">
      <p class="MsoNormal">FASTEH+</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 17.22%; PADDING-TOP: 0cm; HEIGHT: 32.25pt" 
    width="17%">
      <p class="MsoNormal">fpmp,fehp,pfehp</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 15.24%; PADDING-TOP: 0cm; HEIGHT: 32.25pt" 
    vAlign=top width="15%">
      <p class="MsoNormal"><o:p>&nbsp;</o:p></p>
      <p class="MsoNormal">fpmp.doc</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 25.14%; PADDING-TOP: 0cm; HEIGHT: 32.25pt" 
    width="25%">
      <p class="MsoNormal">Faster algorithm and likelihood-based LD 
measures</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 23.64%; PADDING-TOP: 0cm; HEIGHT: 32.25pt" 
    width="23%">
      <p class="MsoNormal">Does not handle missing data, less statistics than 
      EH+</p></td></tr>
  <tr style="HEIGHT: 60pt">
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 17.36%; PADDING-TOP: 0cm; HEIGHT: 60pt" 
    width="17%">
      <p class="MsoNormal">GENECOUNTING&nbsp;</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 17.22%; PADDING-TOP: 0cm; HEIGHT: 60pt" 
    width="17%">
      <p class="MsoNormal">gc,gcp,gcx,pgc</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 15.24%; PADDING-TOP: 0cm; HEIGHT: 60pt" 
    vAlign=top width="15%">
      <p class="MsoNormal"><o:p>&nbsp;</o:p></p>
      <p class="MsoNormal"><o:p>&nbsp;</o:p></p>
      <p class="MsoNormal">gc.txt</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 25.14%; PADDING-TOP: 0cm; HEIGHT: 60pt" 
    width="25%">
      <p class="MsoNormal">Haplotype frequency estimation and reconstruction, 
      flexible in missing data patterns and X chromosome data, 
      haplotype-specific tests</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 23.64%; PADDING-TOP: 0cm; HEIGHT: 60pt" 
    width="23%">
      <p class="MsoNormal">Limited to about 15 SNPs and slow with multiple 
      multiallelic loci with missing data</p></td></tr>
  <tr style="HEIGHT: 30.75pt">
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 17.36%; PADDING-TOP: 0cm; HEIGHT: 30.75pt" 
    width="17%">
      <p class="MsoNormal">HAP</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 17.22%; PADDING-TOP: 0cm; HEIGHT: 30.75pt" 
    width="17%">
      <p class="MsoNormal">hap,mia</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 15.24%; PADDING-TOP: 0cm; HEIGHT: 30.75pt" 
    vAlign=top width="15%">
      <p class="MsoNormal"><o:p>&nbsp;</o:p></p>
      <p class="MsoNormal">hap.txt</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 25.14%; PADDING-TOP: 0cm; HEIGHT: 30.75pt" 
    width="25%">
      <p class="MsoNormal">Large number of SNPs and multiple 
      imputations,&nbsp;missing data, multiallelic loci</p></td>
    <td 
    style="PADDING-RIGHT: 0cm; PADDING-LEFT: 0cm; PADDING-BOTTOM: 0cm; WIDTH: 23.64%; PADDING-TOP: 0cm; HEIGHT: 30.75pt" 
    width="23%">
      <p class="MsoNormal">Possibly sub-optimal solution</p></td></tr></tbody></table>
<ul type="disc">
  <li class="MsoNormal"><b>2LD</b> 
  <ul type="circle">
    <li class="MsoNormal">full name: Two-locus linkage disequilibrium (LD) 
    calculator 
    <li class="MsoNormal">description: <a href="software/2ld.htm">online 
    documentation</a> and a brief <a href="software/2ld.pdf">Adobe PDF</a>/<a 
    href="software/2ld.doc">Word DOC</a> documentation by <a 
    href="http://www.iop.kcl.ac.uk/staff/profile/?go=10610">Dr David Collier</a> 

    <li class="MsoNormal">web: <a href="software/2ld.zip">2ld.zip</a> 
    <li class="MsoNormal">source code language: C 
    <li class="MsoNormal">systems: PC, Unix (SUN Solaris and Linux) 
    <li class="MsoNormal">reference: 
    <ul type="square">
      <li class="MsoNormal"><a href="paper/bi04.pdf">Zhao, J. H. (2004) "2LD, 
      GENECOUNTING and HAP: Computer programs for linkage disequilibrium 
      analysis". Bioinformatics, 20, 1325-1326</a> 
      <li class="MsoNormal"><a href="paper/zapata01.pdf">Zapata C, Carollo C, 
      Rodriguez S. (2001) "Sampling variance and distribution of the D' measure 
      of overall gametic disequlibrium between multiallelic loci". Annals of 
      Human Genetics, 65, 395-406.</a> </li></ul></li></ul>
  <li class="MsoNormal"><b>EH+</b> or <strong>EHPLUS </strong>
  <ul type="circle">
    <li class="MsoNormal">full name: Model-free analysis and permutation tests for 
    allelic association 
    <li class="MsoNormal">version: 1.0, 1.1, 1.2 
    <li class="MsoNormal">descriptions: <a 
    href="software/pm.htm">documentation</a> 
    <li class="MsoNormal">source code language: C 
    <li class="MsoNormal">web: v <a href="software/pm10.zip">1.0</a>, <a 
    href="software/pm11.zip">1.1</a>, <a href="software/pm12.zip">1.2</a>, <a 
    href="software/pm20.zip">2.0</a> 
    <li class="MsoNormal">operating systems: PC and Unix (DEC OSF, Sun Solaris, 
    <span style="COLOR: red">Linux</span>) 
    <li class="MsoNormal"><span lang="PT-BR">reference: </span><a 
    href="paper/hh00.pdf"><span lang="PT-BR">Zhao, J. H., Curtis, D., Sham, P. C. 
    (2000). </span>"Model-free analysis and permutation tests for allelic 
    associations". Human Heredity, 50(2), 133-139</a>. </li></ul>
  <li class="MsoNormal"><b>FASTEH+</b> or <strong>FASTEHPLUS</strong> 
  <ul type="circle">
    <li class="MsoNormal">full name: faster EH+ 
    <li class="MsoNormal">descriptioin: made available in the spirit of Knuth 
    (1998)&nbsp;with <a href="software/fpmp.pdf">documentation</a> (PDF) 
    <li class="MsoNormal">web: <a href="software/fehp.zip">fehp.zip</a> 
    <li class="MsoNormal">source code language: C 
    <li class="MsoNormal">systems: PC, Unix (DEC OSF, SUN Solaris and Linux) 
    <li class="MsoNormal">reference: <a href="paper/hh02.pdf">Zhao, J. H., Sham, 
    P. C. (2002) "Faster allelic association analysis using unrelated subjects". 
    Human Heredity, 53(1), 36-41.</a> </li></ul>
  <li class="MsoNormal"><b>GENECOUNTING</b> 
  <ul type="circle">
    <li class="MsoNormal">full name: gene-counting for haplotype analysis with 
    permutation tests for global association and specific haplotypes, accounting 
    for missing data 
    <li class="MsoNormal"><span lang="FR">description: executable programs with 
    </span><span lang="FR"><a href="software/gc.pdf">documentation</a> 
    <o:p></o:p></span>
    <li class="MsoNormal">web: v <a href="software/gc.zip">1.0</a>, <a 
    href="software/gc11.zip">1.1</a>, <a href="software/gc12.zip">1.2</a>, <a 
    href="software/gc13.zip">1.3</a>, <a href="software/gc20.zip">2.0</a>, <a 
    href="software/gc21.zip">2.1</a>, <a href="software/gc22.zip">2.2</a> 
    <li class="MsoNormal">source code language: C. A version with source code (for 
    autosomal data) and utility programs can be found in <a 
    href="http://www.mds.qmul.ac.uk/statgen/dcurtis/software.html">here</a>. 
    <li class="MsoNormal">systems: PC, Unix (SUN Solaris and Linux) 
    <li class="MsoNormal">reference: 
    <ul type="square">
      <li class="MsoNormal"><a href="paper/bi02.pdf">Zhao, J. H, .Lissarrague, S., 
      Essioux, L., Sham, P. C. (2002) "GENECOUNTING: haplotype analysis with 
      missing genotypes" Bioinformatics, 18(12), 1694-1695</a>. 
      <li class="MsoNormal"><a href="paper/bi04.pdf">Zhao, J. H. (2004) "2LD, 
      GENECOUNTING and HAP: Computer programs for linkage disequilibrium 
      analysis". Bioinformatics, 20, 1325-1326 </a></li></ul></li></ul>
  <li class="MsoNormal"><b>HAP</b> 
  <ul type="circle">
    <li class="MsoNormal">full name: haplotype analysis of polymorphic markers 
    <li class="MsoNormal">descriptioin: extension of <a 
    href="http://www-gene.cimr.cam.ac.uk/clayton/software/snphap.txt">SNPHAP</a> 

    <li class="MsoNormal">web: v <a href="software/hap.zip">0.2.1</a> 
    <li class="MsoNormal">source code language: C 
    <li class="MsoNormal">systems: PC, Unix (SUN Solaris and Linux) 
    <li class="MsoNormal">reference:<a href="paper/bi04.pdf"> Zhao, J. H. (2004) 
    "2LD, GENECOUNTING and HAP: Computer programs for linkage disequilibrium 
    analysis". Bioinformatics, 20, 1325-1326 
</a></li></ul></li></ul></div></body></html>
