<HTML>
<HEAD>
  <META NAME="GENERATOR" CONTENT="Adobe PageMill 3.0 Win">
  <TITLE>KCL: Institute of Psychiatry: Section of Genetic Epidemiology and Biostatistics</TITLE>
  <META NAME="IP" CONTENT="Jing Hua Zhao">
  <META NAME="e-mail" CONTENT="j.zhao@iop.kcl.ac.uk">
  <META NAME="created" CONTENT="**insert**">
  <META NAME="modified" CONTENT="**insert**">
  <META NAME="expires" CONTENT="">
  <META NAME="keywords" CONTENT="**insert**">
  <META NAME="description" CONTENT="Software list">
  <META NAME="copyright" CONTENT="Jing Hua Zhao">
  <META NAME="content_source" CONTENT="Institute of Psychiatry - Psychological Medicine - Genetic Epidemiology and Biostatistics">
  <META NAME="content_destination" CONTENT="College">
  <META NAME="source_path" CONTENT="">
  <LINK REL="STYLESHEET" TYPE="text/css" HREF="/IoP/Includes/kcl_dom.css">
  <LINK REL="STYLESHEET" TYPE="text/css" HREF="/IoP/Includes/kcl_sub.css">
</HEAD>
<BODY BGCOLOR="#ffffff">

<!--NOEDIT-->
<!--#include virtual="/IoP/Includes/kcl_h_sub.html"-->
<!--#include virtual="/IoP/Includes/IoP_GEB_h.html" -->
<!--begin contents-->
<!--/NOEDIT-->

<P><CENTER><TABLE WIDTH="520" BORDER="0" CELLSPACING="0" CELLPADDING="0">
  <TR>
    <TD WIDTH="100%">
    <H3><BR>
    Software Links (Local Copy)</H3>

    <UL>
      <LI>2LD (2-locus LD calculator)
      <LI>3LOCUS (3-locus haplotype analysis)
      <LI><A HREF="http://alleleb.biol.berkeley.edu/~mpn/afbac.html">AFBAC</A>
      (<A HREF="mailto:mpn@alleleb.biol.berkeley.edu">e-mail</A>, Affected
      Family-Based Controls)
      <LI>ALLASS (ALLelic ASSociation)
      <LI>Arlequin (population genetics package)
      <LI><A HREF="ftp://lahmed.stanford.edu/pub/aspex">ASPEX</A> (<A
      HREF="http://www.sco.com/Technology/tcl/Tcl.html">TCL</A>, <A
      HREF="http://plasma-gate.weizmann.ac.il/Xmgr/">Xmgr</A>)
      <LI>ATRIBUTE (path analysis under a generalized single-locus
      model)
      <LI>AUTOZ (computation of the Interval Probability of Autozygosity)
      <LI><A HREF="http://www.mathcs.duq.edu/larget/bambe.html">BAMBE</A>
      (Bayesian Analysis in Molecular Biology and Evolution)
      <LI>BETA (nonparametric linkage using sib-pairs)
      <LI><A HREF="http://www.mbio.ncsu.edu/BioEdit/bioedit.html">BioEdit</A>
      (BIOlogical sequence alignment EDITor)
      <LI>CHRSIM (CHRomosome-based SIMulation)
      <LI>CLUMP (assessing association using Monte Carlo simulation)
      <LI>Cruise (<A HREF="http://www.wpi.edu/~hkim/cruise">a</A>,<A
      HREF="http://www.stat.wisc.edu/~loh/cruise.html">b</A>)
      <LI>CYRILLIC (a pedigree drawing program)
      <LI>dGENE
      <LI>DMAP (Disequilibrium MAPping)
      <LI>ERPA (a program for assessing excessive sharing among relatives)
      <LI>ESPA (Extended Sib Pair Analysis)
      <LI>ETDT (Extended TDT)
      <LI><A HREF="http://wpicr.wpic.pitt.edu/WPICCompGen/">ET-TDT</A>
      (Evolutionary Tree TDT)
      <LI>FASTLINK (Faster LINKAGE)
      <LI>FBAT (Family-Based Association Tests)
      <LI><A HREF="http://www.stat.cmu.edu/~roeder/">FineMap</A> (<A
      HREF="http://www.fmi.uni-passau.de/Graphlet/GTL/">GTL</A>)
      <LI>FISHER (analysis of quantitative traits for general pedigrees)
      <LI>GAS (Genetic Analysis System)
      <LI><A HREF="http://www.mayo.edu/statgen/gassoc/">GASSOC</A>
      (<A HREF="mailto:rowland@mayo.edu">e-mail</A>, Genetic ASSOCiations)
      <LI>GASP (a program for genetic data simulation)
      <LI><A HREF="http://www.stat.cmu.edu/~roeder/">Gcontrol</A> (Association
      using Genomic Control)
      <LI><A HREF="http://lewis.eeb.uconn.edu/lewishome/software.html">GDA</A>
      (Genetic Data Analysis)
      <LI><A HREF="http://www.fhcrc.org/labs/kruglyak/">GeneHunter</A>
      (PLUS, TWOLOCUS, IMPRINTING)
      <LI><A HREF="http://www.cefe.cnrs-mop.fr/">GenePop</A> (population
      genetic software package)
      <LI>GTREE (Gene Tree)
      <LI><A HREF="http://www.cs.wisc.edu/~ghost/gsview/">GhostView</A>
      <LI>GRAPPA (Phylogeny reconstruction from gene orders)
      <LI><A HREF="http://krunch.med.yale.edu/haplo/">HAPLO</A> (HAPLOtype
      analysis)
      <LI><A HREF="http://www.people.fas.harvard.edu/~junliu/index1.html#Computational_Biology">HAPLOTYPER</A>
      <LI><A HREF="http://hmmer.wustl.edu/">HMMER</A> (<A HREF="ftp://ftp.genetics.wustl.edu/pub/eddy/hmmer/">FTP</A>,
      profile hidden Markov models for biological sequence analysis)
      <LI><A HREF="http://www.cbs.dtu.dk/services/HMMgene/">HMMgene</A>
      <LI>HOMOG (HOMOgeneity tests)
      <LI>IntraPop (Software for analysis of mitochondrial DNA polymorphisms)
      <LI><A HREF="http://gnome.agrenv.mcgill.ca/">KIN</A> (Software
      for computing KINship coefficients)
      <LI>Lamarc
      <LI>LINKAGE
      <LI>LIPED (LIkelihood in PEDigrees)
      <LI>MAPMAKER (HOMOZ, SIBS)
      <LI>MAPPOP (Software for selective mapping and bin mapping)
      <LI><A HREF="http://www.megasoftware.net/">MEGA</A> (Molecular
      Evolutionary Genetic Analysis)
      <LI><A HREF="http://watson.hgen.pitt.edu/mega2.html">MEGA2</A>
      (Manipulation Environment for Genetic Analyses)
      <LI><A HREF="http://pegasus.med.ohio-state.edu/">MegaLink</A>
      (interfacing FASTLINK 4.1P, GENEHUNTER 2.0, and SIMPLE)
      <LI>MENDEL (genetic linkage analysis program)
      <LI>MFLINK (Model-free LINKAGE)
      <LI>MIM (Multipoint Interval Mapping)
      <LI><A HREF="ftp://statgen.ncsu.edu/pub/zaykin/">MLD</A> (Multilocus
      Linkage Disequilibrium)
      <LI>MLIKELY (maximum-likelihood estimation)
      <LI>MORGAN (a collection of software, including Markov chain
      Monte Carlo)
      <LI>MSA (Multiple Sequence Alignment)
      <LI>MultiSim (Multipoint Multiple trait Simulation)
      <LI>Mx (structuring equation modeling)
      <LI>PAML (Phylogenic Analysis using Maximum Likelihood)
      <LI>PAP (Pedigree Analysis Package)
      <LI>PATHMIX (Path Analysis for Nuclear Families)
      <LI>PDT (Pedigree Transmission Test)
      <LI>PedCheck (Detecting marker typing incompatibilities in pedigree
      data)
      <LI>Pedraw (drawing pedigrees)
      <LI>PedHunter (a software package used to manage genealogies)
      <LI><A HREF="http://www.myatt.demon.co.uk/index.htm">PEPI</A>
      (Program for EPIdemiology)
      <LI>PdpSys (Pedigree Data Processing System)
      <LI><A HREF="http://www.stats.ox.ac.uk/mathgen/software.html">PHASE</A>
      (a program for reconstructing haplotypes from population genotype
      data)
      <LI><A HREF="http://bozeman.mbt.washington.edu/">Phred/Phrap/consed</A>
      <LI>Phylip (phylogenic analysis)
      <LI>POINTER (Segregation Analysis using POINTERs)
      <LI>POPULUS
      <LI><A HREF="http://icwlin.icw.com/ncss/ncsscalc.zip">Probability
      Calculator</A> (<A HREF="http://www.megasoftware.net/">NCSS</A>)
      <LI><A HREF="http://ibgwww.Colorado.EDU/~lessem/software/qms2.html">QMS</A>
      (DeFries-Fulker Multiple Regression of Sibship QTL Data)
      <LI>QTDT (Quantitative TDT)
      <LI>RELATIVE
      <LI>RELPAIR
      <LI>SAGE (Statistical Analysis for Genetic Epidemiology)
      <LI><A HREF="http://www.cse.ucsc.edu/research/compbio/sam.html">SAM</A>
      (Sequence Alignment and Modeling System)
      <LI><A HREF="http://scop.berkeley.edu/">SCOP</A> (Structural
      classification of proteins, <A HREF="http://scop.wehi.edu.au/scop/">Australia</A>,
      <A HREF="http://scop.mrc-lmb.cam.ac.uk/scop/">Cambridge</A>)
      <LI>SIB-PAIR
      <LI>SIGSCAN (SIGnal SCAN)
      <LI>SimCoal (Simulation of Coalescence)
      <LI>SimIBD (IBD calculation using Simulation)
      <LI>SIMLINK (Simulation program for Linkage)
      <LI>SIMULATE (unconditional simulation for linkage)
      <LI><A HREF="http://pegasus.med.ohio-state.edu/">Simple</A> (analyze
      human pedigree data involving large pedigrees and multiloci)
      <LI>SimWalk2 (Multipoint analyses using Markov chain Monte Carlo
      and simulated annealing)
      <LI>SKUMIX (maximum likelihood to fit a mixture of up to three
      distributions)
      <LI>SLINK (Conditional simulation on pedigrees)
      <LI>SNPHAP (SNP haplotyping)
      <LI>SOLAR (Sequential Oligonenic Linkage Analysis Routine)
      <LI><A HREF="http://www.mrc-bsu.cam.ac.uk/pub/methodology/genetics/">SPLINK</A>
      (Sib Pair LINKage analysis)
      <LI><A HREF="http://hjmuller.bio.uci.edu/~labhome/tools.html">Statistical
      tests and Molecular biology tools</A> (AD Long's lab)
      <LI>STRUCTURE (a model-based clustering method for inferring
      population structure)
      <LI>TABLE (tabulating LINKAGE outputs)
      <LI>TLINKAGE (Two-locus LINKAGE)
      <LI><A HREF="http://www-gene.cimr.cam.ac.uk/clayton/software/">Transmit</A>
      <LI>TreeView
      <LI>VITESSE (Likelihood Calculation on Pedigrees)
      <LI>XDT (Sibling transmission test and quantitative TDT)
    </UL>
</TD>
  </TR>
</TABLE></CENTER></P>

<!--NOEDIT-->
<!--end contents-->
<!--#include virtual="/IoP/Includes/kcl_footer.html"--> 

  <CENTER>
    Last modified: <!--#echo var="LAST_MODIFIED"--> 14/1/2000 by: <A
HREF="mailto:j.zhao@iop.kcl.ac.uk">Jing Hua Zhao</A> 
  </CENTER>

<!--/NOEDIT-->

</BODY>
</HTML>
