<h1 id="hipsci-data-analysis">HIPSCI data analysis</h1>
<p>Two approaches are described below, by PLINK<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> and by bcftools, respectively.</p>
<p>Since the size of each file is quite large (~1.1G for two individuals) an allele frequency filter (e.g., <code>--maf 0.01</code> for PLINK) is appropriate.</p>
<p>Operations through bcftools may be slow but nevertheless the canonical approach.</p>
<h2 id="plink">1. PLINK</h2>
<p>For the case of files in their own directories we could use</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb2-1" title="1"><span class="co"># step 1. to generate PLINK binary (bed/bim/fam) files</span></a>
<a class="sourceLine" id="cb2-2" title="2"><span class="ex">plink</span> --vcf ERZ127238/HPSI1013i-garx_3.wec.gtarray.HumanCoreExome-12_v1_0.imputed_phased.20150604.genotypes.vcf.gz --make-bed --out HPSI1013i-garx_3</a>
<a class="sourceLine" id="cb2-3" title="3"><span class="ex">plink</span> --vcf ERZ127239/HPSI0114i-zapk_1.wec.gtarray.HumanCoreExome-12_v1_0.imputed_phased.20150604.genotypes.vcf.gz --make-bed --out HPSI0114i-zapk_1</a>
<a class="sourceLine" id="cb2-4" title="4"><span class="co"># step 2. to modify .bim to replace variants with missing IDs (&quot;.&quot;) with something like chromosome:position:a1:a2</span></a>
<a class="sourceLine" id="cb2-5" title="5"><span class="co"># step 3. to merge into PLINK</span></a>
<a class="sourceLine" id="cb2-6" title="6"><span class="ex">plink</span> --merge-list ERZ.plist --make-bed --maf 0.01 --out ERZ-maf01</a></code></pre></div>
<p>where step 3 is furnished with <code>ERZ.plist</code> with two entries,</p>
<pre><code>HPSI1013i-garx_3
HPSI0114i-zapk_1</code></pre>
<p>Once all files have been put together, it will be time to perform analysis using information from the .ped files, say.</p>
<h2 id="bcftools">2. bcftools</h2>
<p>The VCF files have been generated by this program, which is available from <a href="https://www.htslib.org/download/" class="uri">https://www.htslib.org/download/</a><a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>.</p>
<p>Suppose we have a file named <code>ERZ.list</code> containing the following list</p>
<pre><code>ERZ127238/HPSI1013i-garx_3.wec.gtarray.HumanCoreExome-12_v1_0.imputed_phased.20150604.genotypes.vcf.gz
ERZ127239/HPSI0114i-zapk_1.wec.gtarray.HumanCoreExome-12_v1_0.imputed_phased.20150604.genotypes.vcf.gz</code></pre>
<p>then we can merge them as follows,</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb6-1" title="1"><span class="ex">bcftools</span> merge --file-list ERZ.list --output ERZ.vcf.gz --output-type z</a></code></pre></div>
<p>The <code>ERZ.vcf.gz</code> then contains data from both VCFs and can be queried as follows,</p>
<pre><code># sample information
bcftools query -l ERZ.vcf.gz

# variant information
bcftools query -f &quot;%CHROM %POS %ID %ALT %REF %INFO \n&quot; ERZ.vcf.gz</code></pre>
<p>As before we could generates a PLINK binary files, keeping variants with MAF above 0.01.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb8-1" title="1"><span class="co"># PLINK</span></a>
<a class="sourceLine" id="cb8-2" title="2"><span class="ex">plink</span> --vcf ERZ.vcf.gz --make-bed --maf 0.01 --out ERZ</a></code></pre></div>
<p>It is possible to replace missing IDs as follows,</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb9-1" title="1"><span class="ex">bcftools</span> annotate --set-id <span class="st">&#39;%CHROM:%POS\_%REF\/%FIRST_ALT&#39;</span> ERZ.vcf.gz -O z -o ERZ-snpid.vcf.gz</a>
<a class="sourceLine" id="cb9-2" title="2"><span class="ex">bcftools</span> index -tf ERZ-snpid.vcf.gz</a></code></pre></div>
<p>A balance needs to be striked on whether to include rsids (%ID) in the new ID.</p>
<h2 id="bash">3. Bash</h2>
<p>This script generates SNP id in format CHROM:POS_REF/ALT (I/D) without bcftools (but index), qctools, vcftools, perl and R<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a> (sorry, folks!).</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb12-1" title="1"><span class="fu">gunzip</span> -c ERZ127238/HPSI1013i-garx_3.wec.gtarray.HumanCoreExome-12_v1_0.imputed_phased.20150604.genotypes.vcf.gz <span class="kw">|</span> <span class="kw">\</span></a>
<a class="sourceLine" id="cb12-2" title="2"><span class="fu">awk</span> <span class="st">&#39;</span></a>
<a class="sourceLine" id="cb12-3" title="3"><span class="st">NR==1,/#CHROM/{print;next}</span></a>
<a class="sourceLine" id="cb12-4" title="4"><span class="st">{</span></a>
<a class="sourceLine" id="cb12-5" title="5"><span class="st">  FS=&quot;\t&quot;;OFS=&quot;\t&quot;</span></a>
<a class="sourceLine" id="cb12-6" title="6"><span class="st">  if(length($4)&gt;1||length($5)&gt;1) if (length($4)&gt;length($5)) {a1=&quot;I&quot;;a2=&quot;D&quot;} else {a1=&quot;D&quot;; a2=&quot;I&quot;} else {a1=$4; a2=$5}</span></a>
<a class="sourceLine" id="cb12-7" title="7"><span class="st">  $3=$1&quot;:&quot;$2&quot;_&quot;a1&quot;/&quot;a2</span></a>
<a class="sourceLine" id="cb12-8" title="8"><span class="st"># $4=a1; $5=a2</span></a>
<a class="sourceLine" id="cb12-9" title="9"><span class="st">  print</span></a>
<a class="sourceLine" id="cb12-10" title="10"><span class="st">}&#39;</span> <span class="kw">|</span> <span class="ex">bgzip</span> -f <span class="op">&gt;</span> ERZ127238-snpid.vcf.gz</a>
<a class="sourceLine" id="cb12-11" title="11"><span class="ex">bcftools</span> index -tf ERZ127238-snpid.vcf.gz</a></code></pre></div>
<p>Optionally when the comment (#) is removed, alleles for an INDEL also become I, D as appropriate.</p>
<p>We could check for multiallelic loci with a function as follows,</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb13-1" title="1"><span class="kw">function</span><span class="fu"> check()</span></a>
<a class="sourceLine" id="cb13-2" title="2"><span class="kw">{</span></a>
<a class="sourceLine" id="cb13-3" title="3">  <span class="fu">gunzip</span> -c <span class="va">${1}</span> <span class="kw">|</span> <span class="kw">\</span></a>
<a class="sourceLine" id="cb13-4" title="4">  <span class="fu">awk</span> <span class="st">&#39;</span></a>
<a class="sourceLine" id="cb13-5" title="5"><span class="st">  NR==1,/#CHROM/{print;next}</span></a>
<a class="sourceLine" id="cb13-6" title="6"><span class="st">  {</span></a>
<a class="sourceLine" id="cb13-7" title="7"><span class="st">    FS=&quot;\t&quot;;OFS=&quot;\t&quot;</span></a>
<a class="sourceLine" id="cb13-8" title="8"><span class="st">    if(length($4)&gt;1||length($5)&gt;1) if (length($4)&gt;length($5)) {a1=&quot;I&quot;;a2=&quot;D&quot;} else {a1=&quot;D&quot;; a2=&quot;I&quot;} else {a1=$4; a2=$5}</span></a>
<a class="sourceLine" id="cb13-9" title="9"><span class="st">    $3=$1&quot;:&quot;$2&quot;_&quot;a1&quot;/&quot;a2</span></a>
<a class="sourceLine" id="cb13-10" title="10"><span class="st">    if(a[$3]++&gt;0) print $1,$2,a1,a2,&quot;multiallelic!&quot;</span></a>
<a class="sourceLine" id="cb13-11" title="11"><span class="st">  }&#39;</span> <span class="op">&gt;</span> <span class="va">${1}</span>.multiallelic</a>
<a class="sourceLine" id="cb13-12" title="12"><span class="kw">}</span></a>
<a class="sourceLine" id="cb13-13" title="13"><span class="ex">check</span> ERZ127238/HPSI1013i-garx_3.wec.gtarray.HumanCoreExome-12_v1_0.imputed_phased.20150604.genotypes.vcf.gz</a></code></pre></div>
<h2 id="association-analysis">4. Association analysis</h2>
<p>We have <code>snp_only.[bed|bim|fam]</code> files and <code>phenos_prot.fam</code> but it contains extra columns</p>
<pre><code>HPSI0114i-iisa_1 SAMEA2398646 0 0 1 24.1204009997205
HPSI0114i-iisa_3 SAMEA2398646 0 0 1 24.4171713024833
HPSI0114i-kolf_2 SAMEA2398402 0 0 1 24.3910589186957
HPSI0114i-vabj_3 SAMEA2398765 0 0 2 22.5672394404035
HPSI0115i-aion_3 SAMEA3182894 0 0 1 24.5824467339929
HPSI0115i-hecn_6 SAMEA3180222 0 0 2 24.2644206030195
HPSI0214i-kucg_2 SAMEA2397923 0 0 1 24.5143044712052
HPSI0214i-rayr_1 SAMEA2399048 0 0 1 24.6123786460459
HPSI0215i-fawm_2 SAMEA3249307 0 0 2 24.454114029875
HPSI0215i-fawm_4 SAMEA3249307 0 0 2 24.4345313709992
...</code></pre>
<p>so we only keep the IDs and protein to proceed with the association analysis</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb15-1" title="1"><span class="fu">cut</span> -d<span class="st">&#39; &#39;</span> -f1,6 phenos_prot.fam <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/_/ /&#39;</span> <span class="op">&gt;</span> phenos_prot.pheno</a>
<a class="sourceLine" id="cb15-2" title="2"><span class="ex">plink</span> --bfile snp_only --pheno phenos_prot.pheno  --linear --out phenos_prot --allow-no-sex</a></code></pre></div>
<p>The <code>--allow-no-sex</code> option is necessary since no sex information is available from <code>snp_only.fam</code>; from these we have <code>phenos_prot.assoc.linear</code> with lines,</p>
<pre><code> CHR              SNP         BP   A1       TEST    NMISS       BETA         STAT            P
   1      1:10583_G/A      10583    A        ADD       64      -0.38       -1.287        0.203
   1      1:30923_G/T      30923    G        ADD       64     0.1602        0.461       0.6464
   1      1:51479_T/A      51479    A        ADD       64    -0.1168      -0.4062        0.686
   1      1:54490_G/A      54490    A        ADD       64    0.06794       0.1514       0.8802
   1      1:54676_C/T      54676    T        ADD       64     0.4491        0.879       0.3828
   1      1:55299_C/T      55299    T        ADD       64    0.03566      0.05709       0.9547
   1      1:57952_A/C      57952    A        ADD       64      0.437       0.7024        0.485
   1      1:61987_A/G      61987    G        ADD       64     0.1729       0.7853       0.4353
   1      1:61989_G/C      61989    C        ADD       64     0.1358       0.6068       0.5462
...</code></pre>
<section class="footnotes">
<hr />
<ol>
<li id="fn1"><h1 id="plink-1">PLINK</h1>
<p>Version 1.9 cannot directly handle the VCF files since some allele labels are too long. One alternative is to use qctool,</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb1-1" title="1"><span class="co"># sudo apt install liblapack-dev</span></a>
<a class="sourceLine" id="cb1-2" title="2"><span class="fu">wget</span> -qO- https://www.well.ox.ac.uk/~gav/resources/qctool_v2.2.0-CentOS_Linux7.8.2003-x86_64.tgz <span class="kw">|</span> <span class="kw">\</span></a>
<a class="sourceLine" id="cb1-3" title="3"><span class="fu">tar</span> xfvz -</a>
<a class="sourceLine" id="cb1-4" title="4"><span class="co"># sudo cp qctool_v2.2.0-CentOS Linux7.8.2003-x86_64/qctool /usr/local/bin</span></a>
<a class="sourceLine" id="cb1-5" title="5"><span class="ex">qctool</span> -filetype vcf -ofiletype binary_ped -g ERZ.vcf.gz -og ERZ</a></code></pre></div>
<a href="#fnref1" class="footnote-back">↩</a></li>
<li id="fn2"><h1 id="installation">Installation</h1>
<p>From WSL2+Ubuntu, run through the following script in order:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb4-1" title="1"><span class="fu">sudo</span> apt update</a>
<a class="sourceLine" id="cb4-2" title="2"><span class="fu">sudo</span> apt install bzip2 libbz2-dev liblzma-dev libcurl4-openssl-dev libssl-dev zlib1g-dev make</a>
<a class="sourceLine" id="cb4-3" title="3"><span class="fu">wget</span> -qO- https://github.com/samtools/bcftools/releases/download/1.16/bcftools-1.16.tar.bz2 <span class="kw">|</span> <span class="kw">\</span></a>
<a class="sourceLine" id="cb4-4" title="4"><span class="fu">tar</span> xjf -</a>
<a class="sourceLine" id="cb4-5" title="5"><span class="bu">cd</span> /bcftools-1.16</a>
<a class="sourceLine" id="cb4-6" title="6"><span class="ex">configure</span></a>
<a class="sourceLine" id="cb4-7" title="7"><span class="fu">make</span></a></code></pre></div>
<a href="#fnref2" class="footnote-back">↩</a></li>
<li id="fn3"><h1 id="notes">Notes</h1>
<p>The size of the header section/data in the body can be determined with</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb10-1" title="1"><span class="bu">export</span> <span class="va">recs=$(</span><span class="fu">gunzip</span> -c <span class="va">$f</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;NR==1,/CHROM/&#39;</span><span class="kw">|</span> <span class="fu">wc</span> -l <span class="kw">|</span> <span class="fu">cut</span> -d<span class="st">&#39; &#39;</span> -f2<span class="va">)</span></a>
<a class="sourceLine" id="cb10-2" title="2"><span class="fu">gunzip</span> -c <span class="va">$f</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;NR==1, $1==&quot;#CHROM&quot; {next}; NF&#39;</span></a></code></pre></div>
<p>A <code>bcftools annotate --set-id</code> equivalent can be achieved by</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb11-1" title="1"><span class="va">$3</span>=<span class="va">$1</span><span class="st">&quot;:&quot;</span><span class="va">$2</span><span class="st">&quot;_&quot;</span><span class="va">$4</span><span class="st">&quot;/&quot;</span><span class="va">$5</span></a></code></pre></div>
<p>without additional work on alleles.<a href="#fnref3" class="footnote-back">↩</a></p></li>
</ol>
</section>
