# HIPSCI data analysis

Two approaches are described below, by PLINK[^plink] and by bcftools, respectively.

Since the size of each file is quite large (~1.1G for two individuals) an allele frequency filter (e.g., `--maf 0.01` for PLINK) is appropriate.

Operations through bcftools may be slow but nevertheless the canonical approach.

## 1. PLINK

For the case of files in their own directories we could use

```bash
# step 1. to generate PLINK binary (bed/bim/fam) files
plink --vcf ERZ127238/HPSI1013i-garx_3.wec.gtarray.HumanCoreExome-12_v1_0.imputed_phased.20150604.genotypes.vcf.gz --make-bed --out HPSI1013i-garx_3
plink --vcf ERZ127239/HPSI0114i-zapk_1.wec.gtarray.HumanCoreExome-12_v1_0.imputed_phased.20150604.genotypes.vcf.gz --make-bed --out HPSI0114i-zapk_1
# step 2. to modify .bim to replace variants with missing IDs (".") with something like chromosome:position:a1:a2
# step 3. to merge into PLINK
plink --merge-list ERZ.plist --make-bed --maf 0.01 --out ERZ-maf01
```

where step 3 is furnished with `ERZ.plist` with two entries,

```
HPSI1013i-garx_3
HPSI0114i-zapk_1
```

Once all files have been put together, it will be time to perform analysis using information from the .ped files, say.

## 2. bcftools

The VCF files have been generated by this program, which is available from <https://www.htslib.org/download/>[^bcftools].

Suppose we have a file named `ERZ.list` containing the following list

```
ERZ127238/HPSI1013i-garx_3.wec.gtarray.HumanCoreExome-12_v1_0.imputed_phased.20150604.genotypes.vcf.gz
ERZ127239/HPSI0114i-zapk_1.wec.gtarray.HumanCoreExome-12_v1_0.imputed_phased.20150604.genotypes.vcf.gz
```

then we can merge them as follows,

```bash
bcftools merge --file-list ERZ.list --output ERZ.vcf.gz --output-type z
```

The `ERZ.vcf.gz` then contains data from both VCFs and can be queried as follows,

```
# sample information
bcftools query -l ERZ.vcf.gz

# variant information
bcftools query -f "%CHROM %POS %ID %ALT %REF %INFO \n" ERZ.vcf.gz
```

As before we could generates a PLINK binary files, keeping variants with MAF above 0.01.

```bash
# PLINK
plink --vcf ERZ.vcf.gz --make-bed --maf 0.01 --out ERZ
```

It is possible to replace missing IDs as follows,

```bash
bcftools annotate --set-id '%CHROM:%POS\_%REF\/%FIRST_ALT' ERZ.vcf.gz -O z -o ERZ-snpid.vcf.gz
bcftools index -tf ERZ-snpid.vcf.gz
```

A balance needs to be striked on whether to include rsids (%ID) in the new ID.

## 3. Bash

This script generates SNP id in format CHROM:POS_REF/ALT (I/D) without bcftools (but index), qctools, vcftools, perl and R[^bash] (sorry, folks!).

```bash
gunzip -c ERZ127238/HPSI1013i-garx_3.wec.gtarray.HumanCoreExome-12_v1_0.imputed_phased.20150604.genotypes.vcf.gz | \
awk '
NR==1,/#CHROM/{print;next}
{
  FS="\t";OFS="\t"
  if(length($4)>1||length($5)>1) if (length($4)>length($5)) {a1="I";a2="D"} else {a1="D"; a2="I"} else {a1=$4; a2=$5}
  $3=$1":"$2"_"a1"/"a2
# $4=a1; $5=a2
  print
}' | bgzip -f > ERZ127238-snpid.vcf.gz
bcftools index -tf ERZ127238-snpid.vcf.gz
```

Optionally when the comment (#) is removed, alleles for an INDEL also become I, D as appropriate.

We could check for multiallelic loci with a function as follows,

```bash
function check()
{
  gunzip -c ${1} | \
  awk '
  NR==1,/#CHROM/{print;next}
  {
    FS="\t";OFS="\t"
    if(length($4)>1||length($5)>1) if (length($4)>length($5)) {a1="I";a2="D"} else {a1="D"; a2="I"} else {a1=$4; a2=$5}
    $3=$1":"$2"_"a1"/"a2
    if(a[$3]++>0) print $1,$2,a1,a2,"multiallelic!"
  }' > ${1}.multiallelic
}
check ERZ127238/HPSI1013i-garx_3.wec.gtarray.HumanCoreExome-12_v1_0.imputed_phased.20150604.genotypes.vcf.gz
```

## 4. Association analysis

We have `snp_only.[bed|bim|fam]` files and `phenos_prot.fam` but it contains extra columns

```
HPSI0114i-iisa_1 SAMEA2398646 0 0 1 24.1204009997205
HPSI0114i-iisa_3 SAMEA2398646 0 0 1 24.4171713024833
HPSI0114i-kolf_2 SAMEA2398402 0 0 1 24.3910589186957
HPSI0114i-vabj_3 SAMEA2398765 0 0 2 22.5672394404035
HPSI0115i-aion_3 SAMEA3182894 0 0 1 24.5824467339929
HPSI0115i-hecn_6 SAMEA3180222 0 0 2 24.2644206030195
HPSI0214i-kucg_2 SAMEA2397923 0 0 1 24.5143044712052
HPSI0214i-rayr_1 SAMEA2399048 0 0 1 24.6123786460459
HPSI0215i-fawm_2 SAMEA3249307 0 0 2 24.454114029875
HPSI0215i-fawm_4 SAMEA3249307 0 0 2 24.4345313709992
...
```

so we only keep the IDs and protein to proceed with the association analysis

```bash
cut -d' ' -f1,6 phenos_prot.fam | sed 's/_/ /' > phenos_prot.pheno
plink --bfile snp_only --pheno phenos_prot.pheno  --linear --out phenos_prot --allow-no-sex
```

The `--allow-no-sex` option is necessary since no sex information is available from `snp_only.fam`; from these we have `phenos_prot.assoc.linear` with lines,

```
 CHR              SNP         BP   A1       TEST    NMISS       BETA         STAT            P
   1      1:10583_G/A      10583    A        ADD       64      -0.38       -1.287        0.203
   1      1:30923_G/T      30923    G        ADD       64     0.1602        0.461       0.6464
   1      1:51479_T/A      51479    A        ADD       64    -0.1168      -0.4062        0.686
   1      1:54490_G/A      54490    A        ADD       64    0.06794       0.1514       0.8802
   1      1:54676_C/T      54676    T        ADD       64     0.4491        0.879       0.3828
   1      1:55299_C/T      55299    T        ADD       64    0.03566      0.05709       0.9547
   1      1:57952_A/C      57952    A        ADD       64      0.437       0.7024        0.485
   1      1:61987_A/G      61987    G        ADD       64     0.1729       0.7853       0.4353
   1      1:61989_G/C      61989    C        ADD       64     0.1358       0.6068       0.5462
...
```

[^plink]:

# PLINK

    Version 1.9 cannot directly handle the VCF files since some allele labels are too long. One alternative is to use qctool,

    ```bash
    # sudo apt install liblapack-dev
    wget -qO- https://www.well.ox.ac.uk/~gav/resources/qctool_v2.2.0-CentOS_Linux7.8.2003-x86_64.tgz | \
    tar xfvz -
    # sudo cp qctool_v2.2.0-CentOS Linux7.8.2003-x86_64/qctool /usr/local/bin
    qctool -filetype vcf -ofiletype binary_ped -g ERZ.vcf.gz -og ERZ
    ```

[^bcftools]:

# Installation

    From WSL2+Ubuntu, run through the following script in order:

    ```bash
    sudo apt update
    sudo apt install bzip2 libbz2-dev liblzma-dev libcurl4-openssl-dev libssl-dev zlib1g-dev make
    wget -qO- https://github.com/samtools/bcftools/releases/download/1.16/bcftools-1.16.tar.bz2 | \
    tar xjf -
    cd /bcftools-1.16
    configure
    make
    ```

[^bash]:

# Notes

    The size of the header section/data in the body can be determined with
    ```bash
    export recs=$(gunzip -c $f | awk 'NR==1,/CHROM/'| wc -l | cut -d' ' -f2)
    gunzip -c $f | awk 'NR==1, $1=="#CHROM" {next}; NF'
    ```
    A `bcftools annotate --set-id` equivalent can be achieved by
    ```bash
    $3=$1":"$2"_"$4"/"$5
    ```
    without additional work on alleles.
