<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<html>
<head>
<meta name="GENERATOR" content="PageBreeze Free HTML Editor (http://www.pagebreeze.com)">
<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1" >
<title>The pilot study</title>
</head>
<body bgcolor="#ffffff">
<h1 align="center">The pilot study</h1>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3" face="Arial">The data in this experiment 
are a sample of 400 controls from the European Prospective Investigation 
of Cancer (EPIC) Norfolk study (</font><a 
href="http://www.srl.cam.ac.uk/epic/"><font color="#606420" 
size=3 face=Arial>http://www.srl.cam.ac.uk/epic/</font></a><font size="3" face="Arial">) each with 
approximately 250k Perlegen (</font><a href="http://www.perlegen.com/"><font 
size=3 face=Arial>http://www.perlegen.com/</font></a><font size="3" face="Arial">) SNPs (the EPIC 400 
study). This is part of a screen sample from a multistage study of breast cancer 
in EPIC. </font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font 
face=Arial></font>&nbsp;</p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><o:p><font 
size=3 face=Arial></font></o:p></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3" face="Arial">The flowchart of our 
implementation is shown in Figure 1. The input data have three sources of 
information, i.e., genotype, map and phenotype. The genotype data contain actual 
genotypes for all individuals in the so-called long format (individual ID, SNP 
ID, and genotype). Map information show position of each SNP by chromosome. 
Phenotypic information is in the usual tabular format (individual ID, sex, body 
mass index and other measurements). These three sources of information are 
merged into a combined dataset for analysis. For screen purpose, only single 
point analysis was conducted. Call rates were obtained, as with HWE tests for 
all SNPs, the results of which were used as an inclusion/exclusion filter of 
SNPs in the regression analysis. The code for SNP information including HWE 
tests is shown as follows,</font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><o:p><font 
size=3 face=Arial></font></o:p></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3" face="Arial"></font>&nbsp;</p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3"><em><font face="Arial">proc sort 
data=epic;<o:p></o:p></font></em></font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3"><em><font face="Arial"><span 
style="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp; </span>by chr 
pos;<o:p></o:p></font></em></font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font 
size=3><em><font face="Arial">run;<o:p></o:p></font></em></font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3"><em><font face="Arial">ods select 
none;<o:p></o:p></font></em></font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3"><em><font face="Arial">proc allele 
data=epic genocol;<o:p></o:p></font></em></font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3"><em><font face="Arial"><span 
style="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp; </span>ods output 
markersumm=ms allelefreq=af genotypefreq=gf;<o:p></o:p></font></em></font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3"><em><font face="Arial"><span 
style="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp; </span>by chr 
pos;<o:p></o:p></font></em></font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3"><em><font face="Arial"><span 
style="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp; </span>var 
a1a2;<o:p></o:p></font></em></font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font 
size=3><em><font face="Arial">run;<o:p></o:p></font></em></font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3" face="Arial"><em>ods select all;</em></font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3">  
<o:p></o:p><font 
face=Arial></font></font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><o:p><font 
size=3 face=Arial></font></o:p></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3" face="Arial">    
             
          
        
            
            
    </font>&nbsp;</p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3" face="Arial">The input data is 
sorted by chromosome (chr) and positions (pos), as input to PROC ALLELE which 
accepts genotype (genocol) and outputs summary information of SNPs (markersumm), 
allele frequencies (allefreq) and genotype frequencies (genotypefreq). The 
outputs are stored in ODS (output delivery system) databases by chromosomes and 
SNP positions, and all outputs for individual SNPs are suppressed (ods select 
none). This shows great simplicity.</font><font size="3" face="Arial">The raw genotype 
data and map information can be used to construct input files for HAPLOVIEW 
(</font><a href="http://www.broad.mit.edu/mpg/haploview/"><font 
size=3 face=Arial>http://www.broad.mit.edu/mpg/haploview/</font></a><font size="3" face="Arial">) for 
visualisation. The SNPs involved can be submitted to ENSEMBL (</font><a 
href="http://www.ensembl.org/index.html"><font 
size=3 face=Arial>http://www.ensembl.org/index.html</font></a><font size="3" face="Arial">) to obtain gene 
annotation.</font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><o:p><font 
size=3 face=Arial></font></o:p></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3" face="Arial">         
      </font>&nbsp;</p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3" face="Arial">Several useful features are notable in this analysis. 
First, we do not require any other software to manage data. In a traditional 
statistical analysis, the data usually takes the so-called wide format, where 
rows indicate sample and columns variables. Since the number of SNPs is quite 
large, it is more sensible to organise genotype data into the so-called long 
format with only a few columns indicating individual IDs, SNP IDs and genotypes. 
Although this requires larger amount of storage but the analysis is considerable 
simpler, for one can perform analysis for each SNP in sequence and store the 
results in a systematic fashion. Second, we were able to take advantage of 
SAS/GENETICS module for HWE and haplotype analysis. Third, all the outputs are 
available as databases for re-use and it is possible to generate data for 
external software programs such as HAPLOVIEW. The facility of result database is 
possible with ODS. Lastly, the SAS programs we developed can run without 
c<st1:personname 
w:st="on">         
      han</st1:personname>ge under Windows. We have created a MS-DOS batch 
file to call SAS from MS-DOS prompt.</font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><o:p><font 
size=3 face=Arial></font></o:p></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3" face="Arial">       
           
             
            
                
              
             
              
</font>&nbsp;</p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font size="3" face="Arial">We have kept the 
comma-separated data in compressed format, to be readily processed by pipe 
mec<st1:personname w:st="on">       
           
             
            
                
              
             
              
han</st1:personname>ism in SAS. We noted that the 
long format repeats individuals IDs, SNP names and uses substantially more 
amount of diskspace compared to the wide format if merged with phenotypic data. 
In this case, the SAS datasets are approximately 30GB if some intermediate 
results are included; the running time is about a day or two on our Intel Linux 
systems with 2GB RAM. Based on this, it is estimated that the full obesity 
project is approximately 30 times larger. But if we spread the task across 
chromosomes on a cluster of 30 nodes the task can be furnished at similar 
speed.</font></p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font 
face=Arial></font>&nbsp;</p>
<p class="MsoNormal" style="MARGIN: 0cm 0cm 0pt"><font 
face=Arial><span 
style="FONT-SIZE: 12pt; FONT-FAMILY: 'Times New Roman'; mso-fareast-font-family: SimSun; mso-ansi-language: EN-GB; mso-fareast-language: ZH-CN; mso-bidi-language: AR-SA"><font face="Arial">Figure 
1. A f<span style="COLOR: black">lowchart of the EPIC 400 Analysis. </span>The raw data 
consist of genotypes, phenotypes and map information, to be merged. Descriptive 
statistics are then obtained, followed by calculation of call rates and HWE, the 
results of which are fed into the regression which assesses statistical 
significance and comparison with theoretical distribution by Q-Q plot. The raw 
data can also be reformatted into HAPLOVIEW input files so that specific region 
in the genome can be visualised.</font> 
            
           
             
         
              
       </span></p></font><span 
style="FONT-SIZE: 12pt; FONT-FAMILY: 'Times New Roman'; mso-fareast-font-family: SimSun; mso-ansi-language: EN-GB; mso-fareast-language: ZH-CN; mso-bidi-language: AR-SA">
<h2 align="left"><span 
style="FONT-SIZE: 12pt; FONT-FAMILY: 'Times New Roman'; mso-fareast-font-family: SimSun; mso-ansi-language: EN-GB; mso-fareast-language: ZH-CN; mso-bidi-language: AR-SA"><span 
style="FONT-SIZE: 12pt; FONT-FAMILY: 'Times New Roman'; mso-fareast-font-family: SimSun; mso-ansi-language: EN-GB; mso-fareast-language: ZH-CN; mso-bidi-language: AR-SA"><span 
style="FONT-SIZE: 12pt; FONT-FAMILY: 'Times New Roman'; mso-fareast-font-family: SimSun; mso-ansi-language: EN-GB; mso-fareast-language: ZH-CN; mso-bidi-language: AR-SA"><span 
style="FONT-SIZE: 12pt; FONT-FAMILY: 'Times New Roman'; mso-fareast-font-family: SimSun; mso-ansi-language: EN-GB; mso-fareast-language: ZH-CN; mso-bidi-language: AR-SA"><font 
face=Arial></font></span></span></span></span> </h2></span>
<p><span 
style="FONT-SIZE: 12pt; FONT-FAMILY: 'Times New Roman'; mso-fareast-font-family: SimSun; mso-ansi-language: EN-GB; mso-fareast-language: ZH-CN; mso-bidi-language: AR-SA"><span 
style="FONT-SIZE: 12pt; FONT-FAMILY: 'Times New Roman'; mso-fareast-font-family: SimSun; mso-ansi-language: EN-GB; mso-fareast-language: ZH-CN; mso-bidi-language: AR-SA"><font face="Arial"><img 
height=720 hspace=0 src="flowchart.jpg" width=960 border=0></font></span></span><font size="1"></p>
<p align="center">&nbsp;</p>
<p align="center">&nbsp;</p></font>
</body>
</html>